src/NServiceBus.Core/Features/Feature.cs

```diff
@@ -64,7 +64,7 @@ protected void Defaults(Action<SettingsHolder> settings)
         /// <summary>
         ///     Called when the features is activated.
         /// </summary>
-        protected internal abstract void Setup(FeatureConfigurationContext context);
+        protected internal abstract IReadOnlyCollection<FeatureStartupTask> Setup(FeatureConfigurationContext context);
```

Comment:
> I wanted to explore a more functional concept where instead of internally register (in therefore intransparently) the tasks will be returned visibly over the return method of the setup call.

The "problem" is, this class then mixes functional and OO design to it's maximum. If we'd like to follow a functional design, we would need to remove the configuration in the constructor and replace that withsome kind of `public FeatureSettings Initialize()` method.